pypsa/io.py

```diff
@@ -870,8 +877,15 @@ def import_components_from_dataframe(network, dataframe, cls_name):
         else:
             if static_attrs.at[k, "type"] == "string":
                 dataframe[k] = dataframe[k].replace({np.nan: ""})
-
-            dataframe[k] = dataframe[k].astype(static_attrs.at[k, "typ"])
+            if dataframe[k].dtype != static_attrs.at[k, "typ"]:
+                if static_attrs.at[k, "type"] == "geometry":
+                    crs = network.meta.pop("_crs", None)
```

Comment:
this doesn't get executed when we import a new network with an empty shape component.

Reply:
Exactly! thanks for raising. Perhaps we should instead go for n._crs object ...

Reply:
yes, that makes more sense as in the current scenario when we do this 
<img width="440" alt="image" src="https://github.com/PyPSA/PyPSA/assets/93286254/3a266645-486e-4d5c-8aed-7394085ff444">
instead of assert {"test":"test"} == {"test" : "test"}
we are getting { "test" : "test , _crs : "....." } == { "test" : "test , _crs : "....." } 
when it is getting passed which might be an issue in some other cases

Reply:
Okay, the more I am thinking about it, the more complicate it seems. I would say for now, we leave it imperfect and don't support tracking of CRS when shape is empty (it is really an edge case). I will remove the CRS setter for now. We should tackle the whole SRID/CRS topic in another PR.